app/src/main/java/org/fossasia/openevent/general/event/EventApi.kt

```diff
@@ -1,13 +1,18 @@
 package org.fossasia.openevent.general.event
 
+import io.reactivex.Flowable
 import io.reactivex.Single
 import retrofit2.http.GET
+import retrofit2.http.Query
 
 interface EventApi {
 
     @GET("events")
     fun getEvents(): Single<List<Event>>
 
+    @GET("events")
+    fun searchEvents(@Query("sort") sort: String, @Query("filter") eventName: String): Flowable<List<Event>>
```

Comment:
Should be single. And are you sure that filtering is working like that? JSONAPI spec says otherwise

Reply:
Yes I am sure. If the users searches "Test" the following endpoint is parsed
https://open-event-api-dev.herokuapp.com/v1/events?sort=name&filter=%5B%7B%22name%22%3A%22name%22%2C%22op%22%3A%22like%22%2C%22val%22%3A%22%25Test%25%22%7D%5D

Also the events are shown in the correct order. I will update the PR soon